pkg/tracker/index_test.go

```diff
@@ -3,40 +3,72 @@
 
 package tracker
 
-// import (
-// 	"fmt"
-// 	"testing"
-
-// 	"github.com/ethereum/go-ethereum/common/hexutil"
-
-// 	"github.com/tellor-io/telliot/pkg/common"
-// 	"github.com/tellor-io/telliot/pkg/db"
-// 	"github.com/tellor-io/telliot/pkg/testutil"
-// )
-// func TestPSR(t *testing.T) {
-// 	ctx, _, cleanup := tcontext.CreateTestContext(t)
-// 	t.Cleanup(cleanup)
-// 	psr, err := BuildIndexTrackers()
-// 	if err != nil {
-// 		testutil.Ok(t, err)
-// 	}
-// 	for idx := range psr {
-// 		err = psr[idx].Exec(ctx)
-// 		psrStr := psr[idx].String()
-// 		if err != nil {
-// 			testutil.Ok(t, errors.New(fmt.SprintF("failed to execute psr: %s %v", psrStr, err)))
-// 		}
-// 	}
-// 	val, err := ctx.Value(common.DBContextKey).(db.DB).Get(fmt.Sprintf("qv_%d", 1))
-// 	if err != nil {
-// 		testutil.Ok(t, err)
-// 	}
-// 	if val == nil {
-// 		testutil.Ok(t, errors.New(Expected a value stored for request ID 1))
-// 	}
-// 	intVal, err := hexutil.DecodeBig(string(val))
-// 	if err != nil {
-// 		testutil.Ok(t, err)
-// 	}
-// 	fmt.Println("DB value", intVal)
-// }
+import (
+	"fmt"
+	"testing"
+
+	"github.com/tellor-io/telliot/pkg/testutil"
+)
+
+type TestDataSource struct {
+	Payload string
+}
+
+func (i TestDataSource) Get() ([]byte, error) {
+	return []byte(i.Payload), nil
+}
+
+type TestCase struct {
+	IndexTracker *IndexTracker
+	Expected     []float64
+}
+
+func TestIndexTracker(t *testing.T) {
+	// Create some test cases.
+	testCases := []TestCase{
+		{
+			IndexTracker: &IndexTracker{
+				Name:       "test1",
+				Identifier: "id1",
+				JSONPath:   "$[0][4]",
+				Source: TestDataSource{
+					Payload: `[[324.34,53453.534,4443.3,45.53453,53.63653]]`,
+				},
+			},
+			Expected: []float64{53.63653},
+		},
+		{
+			IndexTracker: &IndexTracker{
+				Name:       "test2",
+				Identifier: "id2",
+				JSONPath:   `$["test"]["a","b"]`,
+				Source: TestDataSource{
+					Payload: `{"test":{"a":879.54,"b":876.5}}`,
+				},
+			},
+			Expected: []float64{879.54, 876.5},
+		},
+		{
+			IndexTracker: &IndexTracker{
+				Name:       "test1",
+				Identifier: "id1",
+				JSONPath:   "$[0][7,8]",
+				Source: TestDataSource{
+					Payload: "[[768,68,324.34,53453.534,4443.3,45.53453,53.63653,454.534,454.837]]",
+				},
+			},
+			Expected: []float64{454.534, 454.837},
+		},
+	}
+
+	// Test jsonpath parsing per test cases.
+	for _, testCase := range testCases {
+		payload, _ := testCase.IndexTracker.Source.Get()
+		vals, err := testCase.IndexTracker.parsePayload(payload)
```

Comment:
```suggestion
		actual, err := testCase.IndexTracker.parsePayload(payload)
```
not a blocker just personal preference